source/server-discovery-and-monitoring/tests/errors/generate-error-tests.py

```diff
@@ -40,6 +40,7 @@ def write_test(filename, data):
     'ShutdownInProgress': (91,),
     'NotMaster': (10107,),
     'NotMasterNoSlaveOk': (13435,),
+    'LegacyNotPrimary': (10058,),
```

Comment:
All of the spec test changes except for `prefer-error-code.yml` are from the two changes in this file.